api/features/versioning/models.py

```diff
@@ -163,3 +167,212 @@ def clone_to_environment(
+        on_delete=models.CASCADE,
+        related_name="change_sets",
+        blank=True,
+        null=True,
+    )
+    environment_feature_version = models.ForeignKey(
+        "feature_versioning.EnvironmentFeatureVersion",
+        on_delete=models.CASCADE,
+        # Needs to be blank/nullable since change sets
+        # associated with a change request do not yet
+        # have a version
+        blank=True,
+        null=True,
+    )
+
+    feature = models.ForeignKey(
+        "features.Feature",
+        on_delete=models.CASCADE,
+    )
+    live_from = models.DateTimeField(null=True)
+
+    feature_states_to_create = models.TextField(
+        null=True,
+        help_text="JSON blob describing the feature states that should be "
+        "created when the change request is published",
+    )
+    feature_states_to_update = models.TextField(
+        null=True,
+        help_text="JSON blob describing the feature states that should be "
+        "updated when the change request is published",
+    )
+    segment_ids_to_delete_overrides = models.TextField(
+        null=True,
+        help_text="JSON blob describing the segment overrides for which"
+        "the segment overrides should be deleted when the change "
+        "request is published",
+    )
+
+    @hook(BEFORE_CREATE)
+    def add_environment(self):
+        if not self.environment_id:
+            if self.change_request_id:
+                self.environment = self.change_request.environment
+            elif self.environment_feature_version_id:
+                self.environment = self.environment_feature_version.environment
+            else:
+                raise RuntimeError(
+                    "Version change set should belong to either a change request, or a version."
+                )
+
+    def get_parsed_feature_states_to_create(self) -> list[dict[str, typing.Any]]:
+        if self.feature_states_to_create:
+            return json.loads(self.feature_states_to_create)
+        return []
+
+    def get_parsed_feature_states_to_update(self) -> list[dict[str, typing.Any]]:
+        if self.feature_states_to_update:
+            return json.loads(self.feature_states_to_update)
+        return []
+
+    def get_parsed_segment_ids_to_delete_overrides(self) -> list[int]:
+        if self.segment_ids_to_delete_overrides:
+            return json.loads(self.segment_ids_to_delete_overrides)
+        return []
+
+    def publish(self, user: "FFAdminUser") -> None:
+        # TODO:
+        #  - handle API keys
+        from features.versioning.tasks import publish_version_change_set
+
+        kwargs = {"version_change_set_id": self.id, "user_id": user.id}
+        if not self.live_from or self.live_from < timezone.now():
+            publish_version_change_set(**kwargs)
+        else:
+            kwargs["is_scheduled"] = True
+            publish_version_change_set.delay(kwargs=kwargs, delay_until=self.live_from)
+
+    def get_conflicts(self) -> list[Conflict]:
```

Comment:
Note that I've added some unit tests for this method but I have covered a lot more use cases with some integration tests added to the `flagsmith-workflows` repository in the PR [here](https://github.com/Flagsmith/flagsmith-workflows/pull/41).